src/app/modules/color-scheme/color-scheme-popup/color-scheme-popup.component.ts

```diff
@@ -0,0 +1,124 @@
+import { Component, EventEmitter, HostListener, Input, Output } from '@angular/core';
+
+import { ColorScheme, DEFAULT_COLOR_SCHEMES } from '../color-schemes';
+
+/**
+ * Component for the scheme selector popup
+ */
+@Component({
+  selector: 'ccf-color-scheme-popup',
+  templateUrl: './color-scheme-popup.component.html',
+  styleUrls: ['./color-scheme-popup.component.scss']
+})
+export class ColorSchemePopupComponent {
+
+  /**
+   * Controls visibility of the popup
+   */
+  popupVisible = false;
+
+  /**
+   * Current color scheme selected (default is scheme1)
+   */
+  @Input() colorScheme: ColorScheme = DEFAULT_COLOR_SCHEMES[0];
+
+  /**
+   * Schemes available to choose from
+   */
+  @Input() schemeOptions: ColorScheme[] = DEFAULT_COLOR_SCHEMES;
+
+  /**
+   * Current color selected (undefined for gradients)
+   */
+  @Input() color: string | undefined = DEFAULT_COLOR_SCHEMES[0].colors[0];
+
+  /**
+   * Brightness of selected scheme
+   */
+  @Input() brightness: [number, number] = [0, 100];
+
+  /**
+   * Transparency of selected scheme
+   */
+  @Input() transparency = 0;
+
+  /**
+   * Emitted when there is a color scheme change
+   */
+  @Output() colorSchemeChange = new EventEmitter<ColorScheme>();
```

Comment:
Instead of having two outputs, one for scheme and one for color, create an interface, akin to:
```
interface ColorSchemeSelection {
  scheme: ColorScheme;
  index: number;
}
```
and have a single output that emits such objects. This will prevent any mismatch between the currently selected scheme and color.